### PR TITLE
Add .github Setup

### PR DESCRIPTION
**Pull request by grounded042**
This PR sets up `.github` to add code owners as well as linting and testing workflows.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - name: install go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.16
+      - name: checkout code
+        uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          args: -v --timeout=5m --exclude SA1029
```
Comment (grounded042) [inline .github/workflows/lint.yaml:17]: This lint issue would be a breaking change and should wait until a major version bump.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - '*'
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: install go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.16
```
Comment (jfatta) [inline .github/workflows/lint.yaml:18]: I just noticed the go version here is 1.14, any issues with that?
https://github.com/auth0/go-jwt-middleware/blob/575c421b5ffe5e97c21b47ef3e799490d581171f/go.mod#L3

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: install go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.16
```
Comment (jfatta) [inline .github/workflows/test.yaml:20]: same question

**Review by jfatta — APPROVED** [review]
LGTM, just a question reg go version

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (grounded042) [reply .github/workflows/lint.yaml:18]: ah, good call - I'll bump this back to 1.14 for now

Comment (grounded042) [reply .github/workflows/lint.yaml]: @jfatta fixed!

**Review by jfatta — APPROVED** [review]

**Pull request merged**